WalletWasabi.Fluent/ViewModels/Wallets/WalletPageViewModel .cs

```diff
@@ -0,0 +1,93 @@
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using ReactiveUI;
+using WalletWasabi.Fluent.ViewModels.Login;
+using WalletWasabi.Fluent.ViewModels.Navigation;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets;
+
+public partial class WalletPageViewModel : ViewModelBase, IEquatable<WalletPageViewModel>, IComparable<WalletPageViewModel>
+{
+	public Wallet Wallet { get; set; }
+
+	public string Title => Wallet.WalletName;
+
+	private WalletPageViewModel(Wallet wallet)
+	{
+		Wallet = wallet;
+
+		this.WhenAnyValue(x => x.IsSelected)
+			.Where(x => !x && _disposable is { })
+			.Do(_ => _disposable!.Dispose())
+			.Subscribe();
```

Comment:
Instead of using the bang operator and checking `_disposable` have a conditional method invocation:
```suggestion
			.Where(x => !x)
			.Do(_ => _disposable?.Dispose())
			.Subscribe();
```

Or even wrap it into a single do or `Subscribe` accepting a delegate:

```suggestion
			.Subscribe(x =>
			{
				if (!x)
				{
					_disposable?.Dispose();
				}
			});
```

The condition on `_disposable` will be merged with the `!x` check.

Reply:
@YohDeadfall the final version of this code is in UI Decoupling 16 (#10576), which doesn't contain any of that anymore, so I'll close resolve your suggestion here, since fixing it in this PR is only going to cause merge conflict headaches.